examples/bridge-app/esp32/main/main.cpp

```diff
@@ -0,0 +1,452 @@
+    }
+    zclString[0] = static_cast<uint8_t>(len);
+    memcpy(&zclString[1], cString, zclString[0]);
+    return zclString;
+}
+
+// Converted into bytes and mapped the (label, value)
+void EncodeFixedLabel(const char * label, const char * value, uint8_t * buffer, uint16_t length, EmberAfAttributeMetadata * am)
+{
+    char zclOctetStrBuf[kFixedLabelElementsOctetStringSize];
+    uint16_t listCount = 1;
+    _LabelStruct labelStruct;
+
+    labelStruct.label = chip::ByteSpan(reinterpret_cast<const uint8_t *>(label), kFixedLabelElementsOctetStringSize);
+
+    strncpy(zclOctetStrBuf, value, sizeof(zclOctetStrBuf));
+    labelStruct.value = chip::ByteSpan(reinterpret_cast<uint8_t *>(&zclOctetStrBuf[0]), sizeof(zclOctetStrBuf));
+
+    emberAfCopyList(ZCL_FIXED_LABEL_CLUSTER_ID, am, true, buffer, reinterpret_cast<uint8_t *>(&labelStruct), 1);
+    emberAfCopyList(ZCL_FIXED_LABEL_CLUSTER_ID, am, true, buffer, reinterpret_cast<uint8_t *>(&listCount), 0);
+}
+
+EmberAfStatus HandleReadBridgedDeviceBasicAttribute(Device * dev, chip::AttributeId attributeId, uint8_t * buffer,
+                                                    uint16_t maxReadLength)
+{
+    ChipLogProgress(DeviceLayer, "HandleReadBridgedDeviceBasicAttribute: attrId=%d, maxReadLength=%d", attributeId, maxReadLength);
+
+    if ((attributeId == ZCL_REACHABLE_ATTRIBUTE_ID) && (maxReadLength == 1))
+    {
+        *buffer = dev->IsReachable() ? 1 : 0;
+    }
+    else if ((attributeId == ZCL_USER_LABEL_ATTRIBUTE_ID) && (maxReadLength == 32))
+    {
+        ToZclCharString(buffer, dev->GetName(), static_cast<uint8_t>(maxReadLength - 1));
+    }
+    else
+    {
+        return EMBER_ZCL_STATUS_FAILURE;
+    }
+
+    return EMBER_ZCL_STATUS_SUCCESS;
+}
+
+EmberAfStatus HandleReadDescriptorAttribute(uint16_t endpointIndex, chip::AttributeId attributeId, uint8_t * buffer,
+                                            uint16_t maxReadLength)
+{
+    if ((maxReadLength <= kDescriptorAttributeArraySize) && (attributeId < kDescriptorAttributeCount))
+    {
+        memcpy(buffer, &gDescriptorAttrStorage[endpointIndex][attributeId][0], maxReadLength);
+        return EMBER_ZCL_STATUS_SUCCESS;
+    }
+
+    return EMBER_ZCL_STATUS_FAILURE;
+}
+
+EmberAfStatus HandleReadFixedLabelAttribute(Device * dev, EmberAfAttributeMetadata * am, uint8_t * buffer, uint16_t maxReadLength)
+{
+    if ((am->attributeId == ZCL_LABEL_LIST_ATTRIBUTE_ID) && (maxReadLength <= kFixedLabelAttributeArraySize))
+    {
+        EncodeFixedLabel("room", dev->GetLocation(), buffer, maxReadLength, am);
+
+        return EMBER_ZCL_STATUS_SUCCESS;
+    }
+
+    return EMBER_ZCL_STATUS_FAILURE;
+}
+
+EmberAfStatus HandleReadOnOffAttribute(Device * dev, chip::AttributeId attributeId, uint8_t * buffer, uint16_t maxReadLength)
+{
+    ChipLogProgress(DeviceLayer, "HandleReadOnOffAttribute: attrId=%d, maxReadLength=%d", attributeId, maxReadLength);
+
+    if ((attributeId == ZCL_ON_OFF_ATTRIBUTE_ID) && (maxReadLength == 1))
```

Comment:
nit: if you switch the if condition, there may be less nesting:

```cpp
ReturnErrorCodeIf(
    (attributeId != ZCL_ON_OFF_ATTRIBUTE_ID) || (maxReadLength != 1), 
    EMBER_ZCL_STATUS_FAILURE);

*buffer = dev->IsOn() ? 1 : 0;
return EMBER_ZCL_STATUS_SUCCESS;
```

Reply:
done